packages/@o3r/eslint-plugin/src/rules/o3r-widget-tags/o3r-widget-tags.ts

```diff
@@ -0,0 +1,168 @@
+import { createRule } from '../utils';
+
+const o3rWidgetParameterPattern = '^[a-zA-Z0-9-_:.]+$';
+
+export interface O3rWidgetTagsRuleOption {
+  [widgetName: string]: {
+    [paramName: string]: 'string' | 'number' | 'boolean' | 'string[]' | 'number[]' | 'boolean[]';
+  };
+}
+
+type O3rWidgetRuleErrorId = 'notSupportedType' | 'notSupportedParamForType' | 'invalidParamValueType' | 'noParamWithoutWidget' | 'duplicatedWidget' | 'duplicatedParam';
+
+export default createRule<O3rWidgetTagsRuleOption[], O3rWidgetRuleErrorId>({
+  name: 'o3r-widget-tags',
+  meta: {
+    hasSuggestions: false,
+    fixable: undefined,
+    type: 'problem',
+    docs: {
+      description: 'Ensures that @o3rWidget and @o3rWidgetParam are used with correct value',
+      recommended: 'error'
+    },
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: {
+          type: 'object',
+          additionalProperties: false,
+          patternProperties: {
+            [o3rWidgetParameterPattern]: {
+              type: 'string',
```

Comment:
Would be more interesting to be:

```json5
{
  "type": "object",
  "required": ["type"],
  "properties": {
    "required": { "type": "boolnea" },
    "type": {
      "type": "string",
      "enum": [/* ... */]
    }
  }
}
```

To indicate that the paremeter is mandatory (and then provided by the suggestion)

Reply:
I had discussion yesterday with Guillaume(s) and any parameter from a widget will be required

Reply:
It's more to configure rule to decide to make it required in the code than having required in the CMS